### DISCUSSION TITLE
Opsgenie Integration

### DISCUSSION THREAD

**Discussion by Dhrumil-Sentry** [getsentry/sentry#54156] [Launch Feedback]
Users from our [largest customers](https://sentry.io/customers/) have often asked for improvements to our Opsgenie plugin to support additional use cases such as:

- Sending alerts to multiple opsgenie teams/integrations for the same Sentry project-https://github.com/getsentry/sentry/issues/45999
- Sending Metric Alerts to Opsgenie- https://github.com/getsentry/sentry/issues/37347

The lack of these features has been a challenge for enterprise customers that have multiple teams working on the same Sentry project and need to route their alerts to the right users.

We’ve built a new integration with Opsgenie for **Business** and **Enterprise** plan customers to address these requirements and we’re looking for a few early adopters to give us feedback!

# What’s new in this integration

- You can install the integration at an organizational level and can connect multiple Opsgenie accounts to your Sentry organization.
- You will be able to use multiple *Sentry Integration keys* associated with your Opsgenie organization
    - Each team can now add its integration key allowing you to send alerts to different Opsgenie teams for the same Sentry project based on various alert rule filters
- Metric Alerts can also be sent to Opsgenie

# How to enable this integration?

1. Go to your Org’s settings→Integrations
2. Search and Install the _Opsgenie(Integration)_
3. Go to the integration’s configuration and add Opsgenie integration keys (**Note**: You must create a *Sentry Integration* in Opsgenie and add keys associated with those Integrations to this configuration)
![image](https://github.com/getsentry/sentry/assets/109983869/477184a6-f474-4d99-a78a-97be3db26e5c)
![image](https://github.com/getsentry/sentry/assets/109983869/511b91ab-65fa-4f86-90cc-965c90b34843)
4. Once you have added your integration keys to the Opsgenie integration configuration, you can select the Opsgenie account and integration key in Sentry’s Issue and Metric Alert rules
Issue Alerts:
![image](https://github.com/getsentry/sentry/assets/109983869/bcb9d3a3-d6ca-475d-be78-523103d921c7)
Metric Alerts:
![image](https://github.com/getsentry/sentry/assets/109983869/efab175a-1452-446f-b70e-58236a2b3de3)
This will route your alerts to the selected Opsgenie integration.

# Call for Beta Customers- Your feedback is appreciated

We’re actively looking for a few customers who want to use this integration to provide us feedback about the following:

1. Was it easy for you to set up?
2. Is there any Sentry field you would like to see in the Opsgenie Alerts that is not available currently?
3. Sentry is acknowledging the Opsgenie Alert when a metric alert is resolved- Do you prefer a different behavior?

Please comment below with your organization slug or drop us a note at ecosystem-feedback@sentry.io to get early access to this integration 🙏.

**Comment by Dhrumil-Sentry** [comment]
We've made this integration generally available for all Business Plan Customers 🎉

**Reply by wairowe** [reply]
What can Team Plan customers use?

**Reply by Dhrumil-Sentry** [reply]
The legacy plugin we had for Opsgenie is only available to Team Plan customers who were already using the plugin before we built this new integration. Otherwise, customers need to be on the Business plan to use the Opsgenie integration

The plugin experience had several gaps compared to this new integration and we want customers to use this new integration.
